spec/requests/api_registrations_spec.rb

```diff
@@ -16,20 +16,19 @@
       let(:registration_request) { build(:registration_request, competition_id: competition.id, user_id: user.id) }
 
       it 'returns 202' do
-        # post api_v1_registrations_register_path, params: registration_request, headers: headers
-        post api_v1_registrations_register_path, params: registration_request, headers: headers
+        post api_v1_competition_registrations_path(competition_id: competition.id), params: registration_request, headers: headers
         expect(response.body).to eq({ status: "accepted", message: "Started Registration Process" }.to_json)
         expect(response).to have_http_status(:accepted)
       end
 
       it 'enqueues an AddRegistrationJob' do
         expect {
-          post api_v1_registrations_register_path, params: registration_request, headers: headers
+          post api_v1_competition_registrations_path(competition_id: competition.id), params: registration_request, headers: headers
```

Comment:
Bit of a nit, but can we not just pass `competition` here, and in other instances? I've tested locally and it works

Reply:
```suggestion
post api_v1_competition_registrations_path(competition), params: registration_request, headers: headers
```